planetiler-core/src/main/java/com/onthegomap/planetiler/FeatureMerge.java

```diff
@@ -180,8 +189,17 @@ public static List<VectorTile.Feature> mergeLineStrings(List<VectorTile.Feature>
           }
         }
         List<LineString> outputSegments = new ArrayList<>();
+        var i = 0;
         for (Object merged : merger.getMergedLineStrings()) {
-          if (merged instanceof LineString line && line.getLength() >= lengthLimit) {
+          if (merged instanceof LineString line) {
+            // TODO remove debug features comment
+            // Map<String, Object> attrs = new HashMap<>();
+            // attrs.put("idx", i++);
+            // result.add(feature1.copyWithNewGeometry(((LineString) merged).getStartPoint()).copyWithExtraAttrs(attrs));
+            // result.add(feature1.copyWithNewGeometry((LineString) merged).copyWithExtraAttrs(attrs));
+            // attrs.put("end", "yes");
+            // result.add(feature1.copyWithNewGeometry(((LineString) merged).getEndPoint()).copyWithExtraAttrs(attrs));
+
             // re-simplify since some endpoints of merged segments may be unnecessary
             if (line.getNumPoints() > 2 && tolerance >= 0) {
```

Comment:
Because we need to do angle merging after simplification to get connected lines through junctions like this

```
\        /
 x ---- x
/        \
```